src/DotNetty.Codecs/Base64/Base64.cs

```diff
@@ -0,0 +1,379 @@
+                for (; i < calcLength; i += 3)
+                {
+                    if (breakLines)
+                    {
+                        if (charCount == MAX_LINE_LENGTH)
+                        {
+                            destArray[j++] = NEW_LINE;
+                            charCount = 0;
+                        }
+                        charCount += 4;
+                    }
+
+                    destArray[j + 0] = alphabet[(srcArray[i] & 0xfc) >> 2];
+                    destArray[j + 1] = alphabet[((srcArray[i] & 0x03) << 4) | ((srcArray[i + 1] & 0xf0) >> 4)];
+                    destArray[j + 2] = alphabet[((srcArray[i + 1] & 0x0f) << 2) | ((srcArray[i + 2] & 0xc0) >> 6)];
+                    destArray[j + 3] = alphabet[(srcArray[i + 2] & 0x3f)];
+                    j += 4;
+                }
+
+                i = calcLength;
+
+                if (breakLines && (length % 3 != 0) && (charCount == MAX_LINE_LENGTH))
+                {
+                    destArray[j++] = NEW_LINE;
+                }
+                switch (length % 3)
+                {
+                    case 2:
+                        destArray[j + 0] = alphabet[(srcArray[i] & 0xfc) >> 2];
+                        destArray[j + 1] = alphabet[((srcArray[i] & 0x03) << 4) | ((srcArray[i + 1] & 0xf0) >> 4)];
+                        destArray[j + 2] = alphabet[(srcArray[i + 1] & 0x0f) << 2];
+                        destArray[j + 3] = EQUALS_SIGN;
+                        j += 4;
+                        break;
+                    case 1:
+                        destArray[j + 0] = alphabet[(srcArray[i] & 0xfc) >> 2];
+                        destArray[j + 1] = alphabet[(srcArray[i] & 0x03) << 4];
+                        destArray[j + 2] = EQUALS_SIGN;
+                        destArray[j + 3] = EQUALS_SIGN;
+                        j += 4;
+                        break;
+                }
+                //remove last byte if it's NewLine
+                int destLength = destArray[j - 1] == NEW_LINE ? j - 1 : j;
+                return destLength;
+            }
+        }
+        private unsafe static int EncodeUsingGetSet(byte* alphabet, IByteBuffer src, IByteBuffer dest, int offset, int length, bool breakLines)
+        {
+            int srcReaderIndex = src.ReaderIndex;
+            int srcWriterIndex = src.WriterIndex;
+
+            int i = 0;
+            int j = 0;
+            int charCount = 0;
+            int calcLength = offset + length - length % 3;
+            for (i = offset; i < calcLength; i += 3)
+            {
+                if (breakLines)
+                {
+                    if (charCount == MAX_LINE_LENGTH)
+                    {
+                        dest.SetByte(j++, NEW_LINE);
+                        charCount = 0;
+                    }
+                    charCount += 4;
+                }
+
+                dest.SetByte(j + 0, alphabet[(src.GetByte(i) & 0xfc) >> 2]);
```

Comment:
>src.GetByte(i) [](start = 46, length = 14)

pls introduce variables for GetByte(i). .GetByte(i + 1), GetByte(i + 2) calls to avoid unnecessary duplicate access (compiler likely won't do magic here).